volumina/croppingMarkers.py

```diff
@@ -103,28 +141,270 @@ def dataShape(self, shape2D):
         self._width = shape2D[0]
         self._height = shape2D[1]
 
+    @property
+    def cropColor(self):
+        return (self._cropColor)
+    @cropColor.setter
+    def cropColor(self, color):
+        self._cropColor = color
+
+    @property
+    def mouseMoveStartH(self):
+        return self._mouseMoveStartH
+    @mouseMoveStartH.setter
+    def mouseMoveStartH(self, h):
+        self._mouseMoveStartH = h
+
+    @property
+    def mouseMoveStartV(self):
+        return self._mouseMoveStartV
+    @mouseMoveStartV.setter
+    def mouseMoveStartV(self, v):
+        self._mouseMoveStartV = v
+
+    @property
+    def mouseMoveStartCornerH(self):
+        return self._mouseMoveStartCornerH
+    @mouseMoveStartCornerH.setter
+    def mouseMoveStartCornerH(self, h):
+        self._mouseMoveStartCornerH = h
+
+    @property
+    def mouseMoveStartCornerV(self):
+        return self._mouseMoveStartCornerV
+    @mouseMoveStartCornerV.setter
+    def mouseMoveStartCornerV(self, v):
+        self._mouseMoveStartCornerV = v
+
+    def onEditableChanged(self, flag):
+        self._editable = flag
+
     def onExtentsChanged(self, crop_extents_model ):
         crop_extents = crop_extents_model.crop_extents()
         crop_extents.pop(self.axis)
-        
+
         # By default, place cropping lines at 25% and 75%
-        self._vertical0.position = crop_extents[0][0]
-        self._vertical1.position = crop_extents[0][1]
-        self._horizontal0.position = crop_extents[1][0]
-        self._horizontal1.position = crop_extents[1][1]
+        if not (None in crop_extents[0] or None in crop_extents[1]):
+            self._vertical0.position = crop_extents[0][0]
+            self._vertical1.position = crop_extents[0][1]
+            self._horizontal0.position = crop_extents[1][0]
+            self._horizontal1.position = crop_extents[1][1]
+
+            self.prepareGeometryChange()
+
+    def onColorChanged(self, color):
+        self._cropColor = color
         self.prepareGeometryChange()
+        self.update()
 
     def onCropLineMoved(self, direction, index, new_position):
         # Which 3D axis does this crop line correspond to?
-        # (Depends on which orthoview we belong to.)
+        # (Depends on which orthogonal view we belong to.)
+
         axislookup = [[None, 'v', 'h'],
                       ['v', None, 'h'],
                       ['v', 'h', None]]
-        axis_3d = axislookup[self.axis].index( direction[0])
+
+        axis_3d = axislookup[self.axis].index(direction[0])
         crop_extents_3d = self.crop_extents_model.crop_extents()
+
         crop_extents_3d[axis_3d][index] = int(new_position)
         self.crop_extents_model.set_crop_extents( crop_extents_3d )
 
+    def mousePressEvent(self, event):
+
```

Comment:
I still don't understand your comment about multiplications, but I understand that rewriting this function isn't a priority right now.  Just add a detailed docstring to this function and the one below.